### PR TITLE
Run tests with the Clique PoA engine

### LINKED ISSUE #822: Use geth's clique PoA in tests/CI
## Problem Definition

The PoW based `geth` blockchain backend degraded in speed since we initially implemented it. Not only we are bitten by difficulty adjustment for long tests, but also does `geth` internally delay mining if we progress too fast:
```
INFO [07-24|15:38:44] Mining too far in the future             wait=2s
```

PoW furthermore requires DAG preparation (up to 10 minutes) and storage/caching (>=1GB), memory (>=1GB) and a considerable amount of CPU.

Since we run our tests with only miner node and do not test for consensus edge-cases (as of yet), we can switch to [clique PoA](https://github.com/ethereum/EIPs/issues/225), which is supported by `geth` and which we also use as backend for the `raiden smoketest` command.

This will minimize the use of resources and will allow for a rather stable block time of (minimally) around 1 second.

## Solution

I started the work on this in a branch at https://github.com/konradkonrad/raiden/tree/tests_with_clique -- initial results are quite promising: the `long_running` test set takes 200 seconds with PoA instead of 420 seconds with PoW on my machine.

What is missing from my branch is the proper encoding of the authorized miner/signer address (and only this one) in the genesis block ([see here](https://github.com/konradkonrad/raiden/commit/c1b69a9a0ef54fb19d1ae5a073c3be51bfa44883#diff-60a9b084203a3ca241410ac196d6065dR245) and [here](https://github.com/konradkonrad/raiden/commit/c1b69a9a0ef54fb19d1ae5a073c3be51bfa44883#diff-282099ef9fa82f2e3cbda8360e9c1689R139) ) for all fixture paths. 

The simplest approach to this could be to use a fixed miner account across all tests and probably remove all code that allows for multi-(geth-)node test setups.
## Tasklist

- [ ] fix the signer address in fixture genesis blocks
- [ ] remove DAG generation and caching from `.travis.yml`
- [ ] assess test run time for all `geth` backed tests (especially for tests that only use a low number of blocks)

### PR DESCRIPTION
**Pull request by LefterisJP**
Fix #822 

First trying to make sure all tests pass succesfully, and then clean up unneeded mining/DAG code.

### CODE REVIEW

File: raiden/tests/utils/blockchain.py [hunk 1]
```diff
@@ -252,7 +239,8 @@ def geth_create_blockchain(
         config['bootnodes'] = ','.join(node['enode'] for node in nodes_configuration)
 
     all_keys = list(private_keys)
-    all_keys.append(deploy_key)  # needs to be at the end because of the minerthreads keys
+    all_keys.append(deploy_key)
+    all_keys = sorted(set(all_keys))
```
Comment (ulope) [inline raiden/tests/utils/blockchain.py]: Same as above.

File: raiden/tests/fixtures/blockchain.py [hunk 2]
```diff
@@ -226,9 +201,12 @@ def cached_genesis(request, blockchain_type):
 
         genesis_alloc[account_address] = account_alloc
 
+    all_keys = list(private_keys)
```
Comment (hackaugusto) [inline raiden/tests/fixtures/blockchain.py]: Why use a set at all, the private_keys fixture checks for duplicates:

https://github.com/raiden-network/raiden/blob/master/raiden/tests/fixtures/variables.py#L158

File: raiden/tests/fixtures/blockchain.py [hunk 3]
```diff
@@ -226,9 +201,12 @@ def cached_genesis(request, blockchain_type):
...
+    all_keys.append(deploy_key)  # needs to be at the end because of the minerthreads keys
```
Comment (hackaugusto) [inline raiden/tests/fixtures/blockchain.py]: The all_keys collection is sorted in the next line, so the deploy key is not necessarily the last one
